### ISSUE TITLE
Cop idea: File Classes

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12341]
## Description

Don’t nest multi-line classes within classes. Try to have such nested classes each in their own file in a folder named like the containing class.

https://rubystyle.guide/#file-classes

## Behavior

```ruby
# bad

# foo.rb
class Foo
  class Bar
    # 30 methods inside
  end

  class Car
    # 20 methods inside
  end

  # 30 methods inside
end

# good

# foo.rb
class Foo
  # 30 methods inside
end

# foo/bar.rb
class Foo
  class Bar
    # 30 methods inside
  end
end

# foo/car.rb
class Foo
  class Car
    # 20 methods inside
  end
end
```

**Comment by ydakuka** [comment]
https://github.com/rubocop/rubocop-rspec/issues/1691

**Comment by fatkodima** [comment]
Should be disabled by default, if added, because of the lots of false positives. I do not see much benefit in extracting error classes (or some internal small classes) into separate files.
